staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types_jsonschema.go

```diff
@@ -109,6 +109,18 @@ type JSONSchemaProps struct {
 	//      used to identify them. Order is preserved upon merge. The map tag
 	//      must only be used on a list with elements of type object.
 	XListType *string
+
+	// x-kubernetes-map-type annotates an object to further describe its topology.
+	// This extension must only be used when type is object and may have 2 possible values:
+	//
+	// 1) `map`:
```

Comment:
@lavalamp Do we have an opinion on this value? We had initially discussed about "separate", since it's a map whether it's atomic or not.

Reply:
`merge-mode: merge | atomic`

Reply:
I think `merge-mode` is more imperative (verb) rather than declarative (noun), since it's how operations are performed rather than about the data-structure itself.

Reply:
What about: `map-type: independant | atomic`?

Reply:
variable, elastic, flexible, mergeable, patchable, adjustable, adaptable, dynamic

Reply:
I think I like granular, no strong opinions here though

Reply:
> since it's how operations are performed rather than about the data-structure itself

isn't that more accurate? this directive doesn't change the data structure of the map... keys are still the map keys either way. `list-type: set | map` places restrictions on the structure (required unique keys, where keys are the entire value for `list-type:set` or the tuples of named properties for `list-type: map`)

Reply:
No, it's not a directive, it's a type annotation. It fundamentally changes the meaning of the type; each item is separately ownable or only the whole thing may be owned.

granular is a good word for this.

Reply:
I'm fine with `granular`.

Reply:
ok, updated to granular.